app/services/error_details/create_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module ErrorDetails
+  class CreateService < BaseService
+    def initialize(params:, owner:, organization:, integration: nil)
+      super(params:, integration:, owner:, organization:)
+    end
+
+    def call
+      result = super
+      return result if result.error
+
+      res = create_error_details!
+      return res if res&.error
+
+      res
```

Comment:
Here we return `res` in any case. If you wrote it this way because you know you're going to add code after the early return in a following PR, please ignore this comment!

```suggestion
      create_error_details!
```

Reply:
yep, makes sense 🙈